packages/app-insights-client/src/application-insights-client.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { injectable } from 'inversify';
+import { Response } from 'request';
+import * as request from 'request-promise';
+import { EventsQueryOptions } from './events-query-options';
+import { ApplicationInsightsEventsResponse } from './events-query-response';
+import { ApplicationInsightsQueryResponse } from './query-response';
+
+export interface ResponseWithBodyType<T = {}> extends Response {
+    body: T;
+}
+
+@injectable()
+export class ApplicationInsightsClient {
+    private readonly baseUrl = 'https://api.applicationinsights.io/v1/apps';
+    private readonly defaultRequestObject: typeof request;
+    private readonly defaultOptions: request.RequestPromiseOptions = {
+        forever: true,
+        resolveWithFullResponse: true,
+        json: true,
+    };
+
+    constructor(
+        private readonly appId: string,
+        private readonly apiKey: string,
+        private readonly throwOnRequestFailure: boolean = false,
```

Comment:
The `throwOnRequestFailure` logically should not be a part of ctor but func param or class prop. It can be public class prop w/ default value instead